crates/rbuilder/src/live_builder/config.rs

```diff
@@ -68,7 +244,38 @@ impl LiveBuilderConfig for Config {
             MevBoostSlotDataGenerator,
         >,
     > {
-        let live_builder = self.base_config.create_builder(cancellation_token).await?;
+        let submission_config = self
+            .l1_config
+            .submission_config(self.base_config.chain_spec()?)?;
+        info!(
+            "Builder mev boost normal relay pubkey: {:?}",
+            submission_config.signer.pub_key()
+        );
+        info!(
+            "Builder mev boost optimistic relay pubkey: {:?}",
+            submission_config.optimistic_signer.pub_key()
+        );
+        info!(
+            "Optimistic mode, enabled: {}, prevalidate: {}, max_value: {}",
+            submission_config.optimistic_enabled,
+            submission_config.optimistic_prevalidate_optimistic_blocks,
+            format_ether(submission_config.optimistic_max_bid_value),
+        );
+
+        let relays = self.l1_config.relays()?;
+        let sink_factory = RelaySubmitSinkFactory::new(submission_config, relays.clone());
+
+        let payload_event = MevBoostSlotDataGenerator::new(
+            self.l1_config.beacon_clients()?,
+            relays,
+            self.base_config.blocklist()?,
+            cancellation_token.clone(),
+        );
```

Comment:
Shouldn't this be inside L1Config?

Reply:
Not sure about that, this is a function that returns the concrete implementation of the LiveBuilder so it needs to have a reference both to the l1 config and the builder config.